pilot/pkg/config/kube/ingress/conversion.go

```diff
@@ -216,20 +216,22 @@ func ConvertIngressVirtualService(ingress v1beta1.Ingress, domainSuffix string,
 		if f {
 			vs := old.Spec.(*networking.VirtualService)
 			vs.Http = append(vs.Http, httpRoutes...)
-			sort.SliceStable(vs.Http, func(i, j int) bool {
-				r1 := vs.Http[i].Match[0].GetUri()
-				r2 := vs.Http[j].Match[0].GetUri()
-				_, r1Ex := r1.GetMatchType().(*networking.StringMatch_Exact)
-				_, r2Ex := r2.GetMatchType().(*networking.StringMatch_Exact)
-				// TODO: default at the end
-				if r1Ex && !r2Ex {
-					return true
-				}
-				return false
-			})
 		} else {
 			ingressByHost[host] = &virtualServiceConfig
 		}
+
+		// sort routes to meet ingress route precedence requirements
+		// see https://kubernetes.io/docs/concepts/services-networking/ingress/#multiple-matches
+		vs := ingressByHost[host].Spec.(*networking.VirtualService)
+		sort.SliceStable(vs.Http, func(i, j int) bool {
+			r1Len, r1Ex := getMatchURILength(vs.Http[i].Match[0])
+			r2Len, r2Ex := getMatchURILength(vs.Http[j].Match[0])
+			// TODO: default at the end
```

Comment:
shouldn't default be at the end due to len=0?

Reply:
The comment is derived from [here](https://github.com/istio/istio/pull/36847/files#diff-b01367d947f649ce053caa5a9ffda6141cf242cfda73cfd62687fe185603e689L224)

I guess "default" means `pathType: ImplementationSpecific`?